cumulus/client/service/src/lib.rs

```diff
@@ -572,3 +603,137 @@ where
 
 	Err("Stopping following imported blocks. Could not determine parachain target block".into())
 }
+
+async fn parachain_informant<Block: BlockT, Client>(
+	relay_chain_interface: impl RelayChainInterface + Clone,
+	sync_service: Arc<dyn SyncOracle + Send + Sync>,
+	client: Arc<Client>,
+	para_id: ParaId,
+	metrics: Option<ParachainInformantMetrics>,
+) where
+	Client: HeaderBackend<Block> + Send + Sync + 'static,
+{
+	// Backed blocks.
+	let pending_candidates =
+		match pending_candidates(relay_chain_interface.clone(), para_id, sync_service).await {
+			Ok(pending_candidates_stream) => pending_candidates_stream.fuse(),
+			Err(err) => {
+				log::error!("Unable to retrieve pending candidate stream: {err:?}.");
+				return
+			},
+		};
+	let backed_blocks = pending_candidates.flat_map(|(candidates, _, relay_header)| {
+		futures::stream::iter(candidates.into_iter().filter_map(move |candidate| {
+			let relay_header = relay_header.clone();
+			match Block::Header::decode(&mut &candidate.commitments.head_data.0[..]) {
+				Ok(header) => Some((header, relay_header)),
+				Err(e) => {
+					log::warn!("Failed to decode parachain header from backed block: {e:?}");
+					None
+				},
+			}
+		}))
+	});
+	futures::pin_mut!(backed_blocks);
+
+	// Included blocks.
+	let finalized_heads = match finalized_heads(relay_chain_interface.clone(), para_id).await {
+		Ok(finalized_heads_stream) => finalized_heads_stream.fuse(),
+		Err(err) => {
+			log::error!("Unable to retrieve finalized heads stream: {err:?}.");
+			return
+		},
+	};
+	let included_blocks = finalized_heads.filter_map(async |head| {
+		let header = match Block::Header::decode(&mut &head[..]) {
+			Ok(header) => header,
+			Err(e) => {
+				log::warn!("Failed to decode parachain header from finalized block: {e:?}");
+				return None;
+			},
+		};
+		let block_number: U256 = (*header.number()).into();
```

Comment:
You could just do `header.number().is_zero()`.

However in general I am not convinced we even need this extra condition for the genesis block. I assume it would only be printed once anyway? At genesis?